src/main/java/io/cdap/plugin/gcp/spanner/sink/SpannerSink.java

```diff
@@ -95,14 +100,27 @@ public void prepareRun(BatchSinkContext context) throws Exception {
                                                      context.getFailureCollector());
     Configuration configuration = new Configuration();
     configuration.setBoolean(SpannerConstants.IS_PREVIEW_ENABLED, context.isPreviewEnabled());
+    String location;
     if (cmekKeyName != null) {
       configuration.set(SpannerConstants.CMEK_KEY, cmekKeyName.toString());
     }
-    LineageRecorder lineageRecorder = new LineageRecorder(context, config.getReferenceName());
+    // initialize spanner
+    try (Spanner spanner = SpannerUtil.getSpannerService(config.connection.getServiceAccount(),
+                                                         config.connection.isServiceAccountFilePath(),
+                                                         config.connection.getProject())) {
+      // get location for lineage data (location info resides in instance config for spanner)
+      Instance spannerInstance = spanner.getInstanceAdminClient().getInstance(config.getInstance());
+      InstanceConfig instanceConfig = spanner.getInstanceAdminClient().
+        getInstanceConfig(spannerInstance.getInstanceConfigId().getInstanceConfig());
+      location = instanceConfig.getReplicas().get(0).getLocation();
```

Comment:
Is the location guaranteed to be at index 0? Should we default this to some value if it is not there?

Reply:
I made a change to handle a possible null pointer exception in case there is no replica info (unlikely). The location is defaulted to `null` now. If it's passed as `null`, the asset builder will default to `global`.